components/selectors/matching.rs

```diff
@@ -620,6 +620,15 @@ where
     E: Element,
     F: FnMut(&E, ElementSelectorFlags),
 {
+    if cfg!(debug_assertions) {
+        if context.shared.nesting_level == 0 {
+            debug_assert!(!context.shared.in_negation, "How?");
```

Comment:
It seems to me that the whole `if` block is equivalent to
```rust
debug_assert!(context.shared.nesting_level > 0 || !context.shared.in_negation);
```

I cannot think out a case which this single line wouldn't assert but the code here would, and vice verse

Reply:
Oh, nice... Indeed :)